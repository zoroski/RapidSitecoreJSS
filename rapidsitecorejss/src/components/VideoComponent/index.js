import React from 'react';
import Iframe from 'react-iframe';

const VideoComponent = (props) => (
  <div className="container embed-responsive embed-responsive-16by9 my-2">
    <Iframe url={`${props.fields.src.value}`}
        width={`${props.fields.width.value}`}
        height={`${props.fields.height.value}`}
        id="myId"
        className="embed-responsive-item"
        frameborder="0"
        allowfullscreen
    />
  </div>
);

export default VideoComponent;
