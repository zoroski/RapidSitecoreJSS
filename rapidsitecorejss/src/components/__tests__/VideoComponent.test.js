import React from 'react';
import VideoComponent from '../VideoComponent/index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
      const tree = renderer
        .create( < VideoComponent fields = {
            {
              title: {
                value: 'Title',
              },
              src: {
                value: 'https://youtu.be/IeLqT75N8V0',
              },
              width: {
                value: '640',
              },
              height: {
                value: '360',
              },
            }
          }
          />)
          .toJSON(); expect(tree).toMatchSnapshot();
        });