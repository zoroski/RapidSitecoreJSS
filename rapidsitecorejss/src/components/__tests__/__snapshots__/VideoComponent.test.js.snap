// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    Title
  </div>
  <div
    className="row"
  >
    <iframe
      className="myClassname"
      height="360"
      id="myId"
      src="https://youtu.be/IeLqT75N8V0"
      style={
        Object {
          "display": "initial",
          "position": "relative",
        }
      }
      width="640"
    />
  </div>
</div>
`;
